core/base-service/redirector.js

```diff
@@ -1,16 +1,37 @@
 'use strict'
 
+const camelcase = require('camelcase')
 const emojic = require('emojic')
 const Joi = require('joi')
 const BaseService = require('./base')
 const {
   serverHasBeenUpSinceResourceCached,
   setCacheHeadersForStaticResource,
 } = require('./cache-headers')
-const { prepareRoute, namedParamsForMatch } = require('./route')
+const { isValidCategory } = require('./categories')
+const { isValidRoute, prepareRoute, namedParamsForMatch } = require('./route')
 const trace = require('./trace')
 
-module.exports = function redirector({ category, route, target, dateAdded }) {
+const attrSchema = Joi.object({
+  category: isValidCategory,
+  route: isValidRoute,
+  target: Joi.func()
+    .maxArity(1)
```

Comment:
I don't think this thought is all that important (certainly not blocking) but is there a scenario where we'd need a redirector `target` that didn't take in a parameter with the specified route params? If not could we just do `.arity(1)` or would it require adding an accompanying `minArity(1)`

I'm trying to picture a redirector that would have something like the below where the target function wouldn't need an arg. I guess if a service didn't have any route params and/or if we needed a redirect that didn't care about those route params
```js
route: {
  base: 'some/old',
  pattern: ':service/:foo',
},
target: () => 'something/completely/new'
```

Reply:
Yea, we have one of these now 😁 

https://github.com/badges/shields/blob/73d8156f44f2717b4c24a54cff7d7ec591019126/services/endpoint/endpoint-redirect.service.js#L7-L11

Reply:
🤦‍♂️ - my bad for not searching this one